client/src/components/signin/SignIn.js

```diff
@@ -1,13 +1,128 @@
 /* eslint require-jsdoc: "off" */
 import React from 'react';
-import UserInput from '../common/UserInput';
+import { Link } from 'react-router';
+import { connect } from 'react-redux';
+import toastr from 'toastr';
+import docImage from '../../images/dms.jpg';
+import { login } from '../../actions/userActions';
 
 class SigninForm extends React.Component {
+  /**
+   *
+   * @param {Object} props
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: {},
+      errors: {}
+    };
+
+    this.onClick = this.onClick.bind(this);
+    this.onChange = this.onChange.bind(this);
+  }
+
+  /**
+   *
+   * @param {*} event
+   * @returns {*} click action
+   */
+  onClick(event) {
+    event.preventDefault();
+    if (this.isValid()) {
+      this.setState({ errors: {}, isLoading: true });
+      this.props.login(this.state.user).then(
+        res => this.context.router.push('/dashboard'),
+        err => toastr.error('Incorrect login details')
+      );
+    }
+  }
+
+  /**
+   *
+   * @param {*} event
+   * @returns {*} action
+   */
+  onChange(event) {
+    const field = event.target.name;
+    const user = this.state.user;
+    user[field] = event.target.value;
+    this.setState({ user });
+  }
+
+  /**
+   *
+   * @param {*} null
+   * @returns {Boolean} Boolean validating the form
+   */
+  isValid() {
+    let formIsValid = true;
+    const errors = {};
+    if (!this.state.user.email || !this.state.user.password) {
+      errors.error = 'Please fill the required fields';
+      formIsValid = false;
+    }
+    this.setState({ errors });
+    return formIsValid;
+  }
+
+  /**
+   * Renders the UserInput component
+   * @param {*} null
+   * @returns {*} rendered JSX
+   */
   render() {
     return (
-      <UserInput />
+      <div className="row">
```

Comment:
**Nitpick**
This should be a container component. So, you move this markup to a `signupForm` presentational component